### ISSUE TITLE
generate module options docs

### ISSUE THREAD

**Issue by hall** [hall/kubenix#1]
Would be nice to create a static site that lists module options.

I've got some WIP stuff on the linked branch (see sidebar) but there's currently a possible infinite recursion error that causes a segfault when evaluating the generated Kubernetes modules (e.g., `./modules/generated/v1.24.nix`).

Preview of the current state, only generating options for the `docker` module:
![image](https://user-images.githubusercontent.com/32495955/186981377-fc1d4f05-5f63-48d0-8b21-d59ca09f8fb4.png)

**Comment by hall** [comment]
Also had to apply this diff for [the eval to succeed](https://github.com/hall/kubenix/runs/8043685586?check_suite_focus=true):

```diff
diff --git a/modules/docker.nix b/modules/docker.nix
index c033dff..713db70 100644
--- a/modules/docker.nix
+++ b/modules/docker.nix
@@ -34,13 +34,13 @@ in {
           name = mkOption {
             description = "Desired docker image name";
             type = types.str;
-            default = builtins.unsafeDiscardStringContext config.image.imageName;
+            default = "";
           };
 
           tag = mkOption {
             description = "Desired docker image tag";
             type = types.str;
-            default = builtins.unsafeDiscardStringContext config.image.imageTag;
+            default = "";
           };
 
           registry = mkOption {
```

Presumably, we need to pull in some upstream config as those values don't appear to be defined in this repo.

**Comment by hall** [comment]
Alright, getting late but I've narrowed the `k8s` module evaluation build failure to the CRD type and was able to successfully build with the following workaround:

```diff
diff --git a/modules/generated/v1.24.nix b/modules/generated/v1.24.nix
index 5bb237c..78b80da 100644
--- a/modules/generated/v1.24.nix
+++ b/modules/generated/v1.24.nix
@@ -17270,7 +17270,8 @@ in
       };
       "apiextensions.k8s.io"."v1"."CustomResourceDefinition" = mkOption {
         description = "CustomResourceDefinition represents a resource that should be exposed on the API server.  Its name MUST be in the format <.spec.name>.<.spec.group>.";
-        type = (types.attrsOf (submoduleForDefinition "io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinition" "customresourcedefinitions" "CustomResourceDefinition" "apiextensions.k8s.io" "v1"));
+        type = types.attrs;
         default = { };
       };
       "apiregistration.k8s.io"."v1"."APIService" = mkOption {
@@ -17335,11 +17336,11 @@ in
         type = (types.attrsOf (submoduleForDefinition "io.k8s.api.batch.v1.CronJob" "cronjobs" "CronJob" "batch" "v1"));
         default = { };
       };
       "customResourceDefinitions" = mkOption {
         description = "CustomResourceDefinition represents a resource that should be exposed on the API server.  Its name MUST be in the format <.spec.name>.<.spec.group>.";
-        type = (types.attrsOf (submoduleForDefinition "io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinition" "customresourcedefinitions" "CustomResourceDefinition" "apiextensions.k8s.io" "v1"));
+        type = types.attrs;
         default = { };
       };
       "daemonSets" = mkOption {
         description = "DaemonSet represents the configuration of a daemon set.";
         type = (types.attrsOf (submoduleForDefinition "io.k8s.api.apps.v1.DaemonSet" "daemonsets" "DaemonSet" "apps" "v1"));
```

**Comment by hall** [comment]
CRD type evaluation fixed (see #2). Last piece left is to undo that diff in the `docker` module above.
